gateway/ln-gateway/src/lib.rs

```diff
@@ -1149,19 +1147,67 @@ impl Gateway {
     /// `per_federation_routing_fees` is changed, the Gateway will only
     /// re-register with the specified federation.
     pub async fn handle_set_configuration_msg(
+        &self,
+        payload: SetConfigurationPayload,
+    ) -> Result<()> {
+        let mut dbtx = self.gateway_db.begin_transaction().await;
+
+        let new_gateway_config = self.create_new_gateway_config(&payload).await?;
+        dbtx.insert_entry(&GatewayConfigurationKey, &new_gateway_config)
```

Comment:
Then this could be `dbtx.store_gateway_config` etc.

Reply:
Needs to be the same `dbtx` though that gets committed at the bottom, so you previous comment is necessary for this